subprojects/core-api/src/main/java/org/gradle/api/component/ConfigurationVariantDetails.java

```diff
@@ -38,6 +38,16 @@
      */
     void skip();
 
+    /**
+     * Provides information about the optional status of this added configuration variant.
+     *
+     * <ul>
+     *     <li>For the Maven world, this means that dependencies will be declared as {@code optional}.</li>
+     *     <li>For the Ivy world, this means that configuration marked optional will not be extended by the {@code default} configuration.</li>
+     * </ul>
+     */
+    void markOptional();
```

Comment:
:x: Let's be clearer that it's a mapping to Maven/Ivy issue, by calling it `mapToOptional()`.